core/arch/aarch64/proc.c

```diff
@@ -88,11 +88,35 @@ get_processor_specific_info(void)
     cpu_info.features.flags_aa64mmfr1 = isa_features[AA64MMFR1];
     cpu_info.features.flags_aa64dfr0 = isa_features[AA64DFR0];
 
-#        if !defined(DR_HOST_NOT_TARGET) && defined(SVE)
-    /* TODO i#3044: Vector length will be set by reading value from h/w. */
-    CLIENT_ASSERT(false, "TODO i#3044: SVE requires initialisation of vector length!");
-#        elif !defined(STANDALONE_DECODER) || defined(DR_HOST_NOT_TARGET)
-    /* Set SVE vector length for unit tests. */
+    /* The SVE vector length is set to:
+     * - A value read from the host hardware.
+     * or:
+     * - 32 bytes, 256 bits.
+     * Which of the above depends on:
+     * - SVE or non-SVE AArch64 or x86 host h/w.
+     * and:
+     * - Release or development test build.
+     */
+#        if !defined(DR_HOST_NOT_TARGET)
+    if (proc_has_feature(FEATURE_SVE)) {
+#            if !defined(BUILD_TESTS)
+        uint64 vl;
+        /* TODO i#5365: Ideally this should be generated by INSTR_CREATE_rdvl()
+         * and executed at startup time with other initialisation code.
+         */
+        asm(".inst 0x04bf5020\n" /* rdvl x0, #1 */
```

Comment:
The latest toolchains do support SVE instructions but we don't build with SVE enabled: i.e. not `-march=armv8-a+sve`, so that we can run a single DynamoRIO release on both SVE and non-SVE h/w. This `RDVL` is raw hex to avoid errors when building for base v8.

Reply:
That makes sense.  Maybe a brief comment to avoid confusion from future readers?

Reply:
Done. Improved comments to describe the reason why we're using raw hex until the `TODO` is implemented.